content/en/blog/gitops-your-falco-rules/index.md

```diff
@@ -0,0 +1,805 @@
+```
+
+The first container seems to be based on the `falco-no-driver` image (the driver is loaded separately), and the second one has the name `falcoctl-artifact-follow`. If you went through the previous section, you'll see the name of the container matches the service name in Linux.
+
+Now we need to look for the configuration file, We saw in the previous section its name was `/etc/falcoctl/falcoctl.yaml`. Let's start extracting the volumes on the `falcoctl-artifact-follow` container:
+
+```
+$ kubectl -n falco get pod falco-8dv99 -o json | jq '.spec.volumes[]'
+[
+... output omitted ...
+{
+  "configMap": {
+    "defaultMode": 420,
+    "items": [
+      {
+        "key": "falcoctl.yaml",
+        "path": "falcoctl.yaml"
+      }
+    ],
+    "name": "falco-falcoctl"
+  },
+}
+... output omitted ...
+]
+```
+Now that we have the configMap name, we can proceed to update its configuration.
+
+#### Customing the Falcoctl ConfigMap
+It is a good practice to observe what normal logs look like before changing a program to a newer configuration. So just to keep track of it:
+
+```
+$ kubectl -n falco logs falco-8dv99 -c falcoctl-artifact-follow 
+
+INFO: Retrieving versions from Falco (timeout 2m0s) ...
+INFO: Successfully retrieved versions from Falco ...
+INFO: Reading all configured index files from "/root/.config/falcoctl/indexes.yaml"
+WARN: No configured index. Consider to configure one using the 'index add' command.
+INFO: Creating follower for "falco-rules:0", with check every 6h0m0s
+... output omitted...
+```
+
+Nothing new, it follows the `falco-rules:0` artifact and checks every 6 hours. Same as in the previous section.
+
+Here we have the default configuration installed by Helm when deploying `falco` and `falcoctl`:
+
+```
+$ kubectl -n falco get cm falco-falcoctl -o yaml
+... output omitted ...
+data:
+  falcoctl.yaml: |-
+    artifact:
+      allowedTypes:
+      - rulesfile
+      follow:
+        every: 6h
+        falcoversions: http://localhost:8765/versions
+        pluginsDir: /plugins
+        refs:
+        - falco-rules:0
+        rulesfilesDir: /rulesfiles
+      install:
+        pluginsDir: /plugins
+        refs:
+        - falco-rules:0
+        resolveDeps: false
+        rulesfilesDir: /rulesfiles
+    indexes:
+    - name: falcosecurity
+      url: https://falcosecurity.github.io/falcoctl/index.yaml
+... output omitted ...
+```
+and here are the changes that we want to do on it so it checks every 2 minutes, as before, and for a newer OCI artifact. We are not updating the rulesfilesDir parameter here since the subdirectory `etc/falco/rules.d/` doesn't exist on the `falco` container.
+```
+...
+    artifact:
+      ...
+      follow:
+        every: 2m
+        refs:
+        - ghcr.io/vjjmiras/cool-falco-ruleset/custom-rules:main
+...
+```
+
+Once we save those changes, we can delete the older Pod and see the new one starting with the newest configuration:
```

Comment:
Can you explicitly notice what to change in the Helm values.yaml and to run `helm upgrade`, the restart of the Pods will be automatic then.